src/site/content/en/blog/google-search-sw/index.md

```diff
@@ -0,0 +1,463 @@
+an update check is made after 25 minutes has passed, the server can respond
+efficiently with an [HTTP 304](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/304)
+response if there haven't been any updates to the service worker deployed in the
+interim.
+
+While some interactions within the Google Search web app do use single page
+app-style navigations (i.e. via the
+[History API](https://developer.mozilla.org/en-US/docs/Web/API/History_API#Adding_and_modifying_history_entries)),
+for the most part, Google Search is a traditional web app that uses "real"
+navigations. This comes into play when the team decided that it would be
+effective to use two options that accelerate the service worker update
+lifecycle:
+[`clients.claim()`](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#clientsclaim)
+and
+[`skipWaiting()`](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#skip_the_waiting_phase).
+Clicking around Google Search's interface will generally end up navigating to
+brand new HTML documents, and calling `skipWaiting()` ensures that an
+updated service worker will get a chance to handle those new navigation requests
+immediately after installation. Similarly, calling `clients.claim()`
+means that the updated service worker will get a chance to start controlling any
+open Google Search pages that are uncontrolled, following service worker
+activation.
+
+The approach that Google Search went with isn't necessarily a solution that will
+work for everyone—it was the result of carefully A/B testing various
+combinations of serving options until they found what worked best for them.
+Developers whose backend infrastructure allow them to deploy updates more
+quickly might prefer that the browser check for an updated service worker script
+as frequently as possible, by
+[always ignoring the HTTP cache](https://developers.google.com/web/updates/2018/06/fresher-sw#whats_changing).
+If you're building a single page app that users will might keep open for a long
+period of time, using `skipWaiting()` is probably not the right choice for
+you—you
+[risk running into cache inconsistencies](https://developers.google.com/web/fundamentals/primers/service-workers/lifecycle#skip_the_waiting_phase)
+if you allow the new service worker to activate while there are long-lived
+clients.
+
+## Key Takeaways
+
+### By default, service workers aren't performance neutral
+
+Adding a service worker to your web app means inserting an additional piece of
+JavaScript that needs to be loaded and executed before your web app gets
+responses to its requests. If those responses end up coming from a local cache
+rather than from the network, then the overhead of running the service worker
+will usually be negligible in comparison to the performance win from going
+cache-first. But if you know that your service worker will always have to
```

Comment:
Again, remove 'will' - if yo know that your service worker always has to consult the network'''